parser/auth/auth.go

```diff
@@ -41,18 +41,24 @@ func (user *UserIdentity) Restore(ctx *format.RestoreCtx) error {
 }
 
 // String converts UserIdentity to the format user@host.
+// It defaults to providing the AuthIdentity (the matching entry in priv tables)
```

Comment:
NOTE: This totally change the semantic of the old API, and all the code referencing of the old API might be affected.

```
AuthIdentityString() -> String()
String() -> LoginString()
```

I can't say the new API is better or necessay (Go also discourage public API change), but as long as it fix more cases and do not introduce regression, it looks good to me.